src/minetest.cpp

```diff
@@ -166,6 +163,31 @@ bool Minetest::runMod(EditorState *state, const std::string &world)
 	FileFormat *writer = getFromType(FILE_FORMAT_LUA, state);
 	save_file(writer, state, mod_to + "init.lua");
 
+	// Build argument list
+	char *argv[9];
+	argv[0] = (char*)(minetest_exe).c_str();
+	argv[1] = "--worldname";
+	argv[2] = (char*)world.c_str();
+	argv[3] = "--name";
+	argv[4] = "tester";
+	argv[5] = "--address";
+	argv[6] = "";
+	argv[7] = "--go";
+	argv[8] = NULL;
```

Comment:
You should be able to fix the warning you're getting by changing the type to `char *const argv[9]`.
Also, a nicer way to do it:
```C++
char *const argv[] = {
	minetest_exe.c_str(),
	"--worldname", world.c_str(),
	"--name", "tester",
	"--address", "",
	"--go",
	NULL,
};
```